tests/pytests/test_info_modules.py

```diff
@@ -1989,3 +1989,224 @@ def test_coord_high_priority_pending_jobs_hybrid(self):
+  conn.execute_command('HSET', 'tag:1', 'tag', 'value1,value2')
+  metrics = get_field_metrics()
+  env.assertEqual(metrics['tag'], 1, message="After 1 tag doc")
+
+  conn.execute_command('HSET', 'tag:2', 'tag', 'value3')
+  metrics = get_field_metrics()
+  env.assertEqual(metrics['tag'], 2, message="After 2 tag docs")
+
+  # 3. Test NUMERIC field indexing
+  env.expect('FT.CREATE', 'idx_num', 'PREFIX', 1, 'num:', 'SCHEMA', 'n', 'NUMERIC').ok()
+  waitForIndex(env, 'idx_num')
+
+  conn.execute_command('HSET', 'num:1', 'n', '42')
+  metrics = get_field_metrics()
+  env.assertEqual(metrics['numeric'], 1, message="After 1 numeric doc")
+
+  conn.execute_command('HSET', 'num:2', 'n', '100')
+  metrics = get_field_metrics()
+  env.assertEqual(metrics['numeric'], 2, message="After 2 numeric docs")
+
+  # 4. Test GEO field indexing
+  env.expect('FT.CREATE', 'idx_geo', 'PREFIX', 1, 'geo:', 'SCHEMA', 'g', 'GEO').ok()
+  waitForIndex(env, 'idx_geo')
+
+  conn.execute_command('HSET', 'geo:1', 'g', '13.361389,52.519444')  # Berlin
+  metrics = get_field_metrics()
+  env.assertEqual(metrics['geo'], 1, message="After 1 geo doc")
+
+  conn.execute_command('HSET', 'geo:2', 'g', '-0.127758,51.507351')  # London
+  metrics = get_field_metrics()
+  env.assertEqual(metrics['geo'], 2, message="After 2 geo docs")
+
+  # 5. Test VECTOR field indexing
+  env.expect('FT.CREATE', 'idx_vec', 'PREFIX', 1, 'vec:',
+             'SCHEMA', 'v', 'VECTOR', 'FLAT', '6',
+             'TYPE', 'FLOAT32', 'DIM', '2', 'DISTANCE_METRIC', 'L2').ok()
+  waitForIndex(env, 'idx_vec')
+
+  vec1 = np.array([1.0, 0.0]).astype(np.float32).tobytes()
+  vec2 = np.array([0.0, 1.0]).astype(np.float32).tobytes()
+
+  conn.execute_command('HSET', 'vec:1', 'v', vec1)
+  metrics = get_field_metrics()
+  env.assertEqual(metrics['vector'], 1, message="After 1 vector doc")
+
+  conn.execute_command('HSET', 'vec:2', 'v', vec2)
+  metrics = get_field_metrics()
+  env.assertEqual(metrics['vector'], 2, message="After 2 vector docs")
+
+  # 6. Test multiple fields in same document
+  env.expect('FT.CREATE', 'idx_multi', 'PREFIX', 1, 'multi:',
+             'SCHEMA', 't', 'TEXT', 'tag', 'TAG', 'n', 'NUMERIC').ok()
+  waitForIndex(env, 'idx_multi')
+
+  # Store current counts
+  prev_metrics = get_field_metrics()
+
+  conn.execute_command('HSET', 'multi:1', 't', 'hello', 'tag', 'mytag', 'n', '1')
+  metrics = get_field_metrics()
+  env.assertEqual(metrics['text'], prev_metrics['text'] + 1,
+                  message="Multi-field doc increments text")
+  env.assertEqual(metrics['tag'], prev_metrics['tag'] + 1,
+                  message="Multi-field doc increments tag")
+  env.assertEqual(metrics['numeric'], prev_metrics['numeric'] + 1,
+                  message="Multi-field doc increments numeric")
+
+  # 7. Test double counting with overlapping indexes
+  # Create another text index that will also match 'text:*' docs
+  env.expect('FT.CREATE', 'idx_text2', 'PREFIX', 1, 'text:', 'SCHEMA', 't', 'TEXT').ok()
```

Comment:
consider adding an index with more than one field of the same type then:
1. a doc that matches only one of them
2. a doc that contains both
3. a doc that contains none (not necessary)